src/action/macos/encrypt_apfs_volume.rs

```diff
@@ -57,7 +59,11 @@ impl EncryptApfsVolume {
             // The user has a password matching what we would create.
             if planned_create_apfs_volume.state == ActionState::Completed {
```

Comment:
I wonder if this is still true in the case of this "enterprise edition" -- since we add specific access to another binary, isn't it possible that if we take the current volume as-is, we'll not have that access and things will break?

(I think all of these `completed` branches need to be audited to ensure that this is OK)

Reply:
All the other branches look good to me ... one option is to unconditionally add our program to the access list. I don't know how else we can detect a fault here.